addons/l10n_be/models/res_company.py

```diff
@@ -0,0 +1,13 @@
+from odoo import api, fields, models, tools, _
+
+
+class ResCompany(models.Model):
+    _inherit = 'res.company'
+
+    company_registry = fields.Char(compute='_compute_company_registry', store='True', readonly=False)
+
+    @api.depends('vat')
+    def _compute_company_registry(self):
+        for company in self:
+            if company.country_id.code == 'BE' and company.vat:
+                company.company_registry = company.vat[2:]
```

Comment:
for company in self:
            company_registry = self.company_registry
            if company.country_id.code == 'BE' and company.vat:
                company_registry = company.vat[2:]
            self..company_registry = company_registry 

please also add some comments to say what you do and why you do it.

thanks

Reply:
Added # If a belgian company has a VAT number then it's company registry is it's VAT Number (without country code).
Good for you ?

Reply:
Added comments, check for preexising value & vat number validity.